tap_linkedin_ads/client.py

```diff
@@ -27,65 +26,99 @@ class LinkedInUnauthorizedError(LinkedInError):
-    return ERROR_CODE_EXCEPTION_MAPPING.get(error_code, LinkedInError)
+    400: {
+        "raise_exception": LinkedInBadRequestError,
+        "message": "The request is missing or has a bad parameter."
+    },
+    401: {
+        "raise_exception": LinkedInUnauthorizedError,
+        "message": "Invalid authorization credentials."
+    },
+    403: {
+        "raise_exception": LinkedInForbiddenError,
+        "message": "User does not have permission to access the resource."
+    },
+    404: {
+        "raise_exception": LinkedInNotFoundError,
+        "message": "The resource you have specified cannot be found."
+    },
+    405: {
+        "raise_exception": LinkedInMethodNotAllowedError,
+        "message": "The provided HTTP method is not supported by the URL."
+    },
+    411: {
+        "raise_exception": LinkedInLengthRequiredError,
+        "message": "The server refuses to accept the request without a defined Content-Length header."
+    },
+    429: {
+        "raise_exception": LinkedInRateLimitExceeededError,
+        "message": "API rate limit exceeded, please retry after some time."
+    },
+    500: {
+        "raise_exception": LinkedInInternalServiceError,
+        "message": "An error has occurred at LinkedIn's end."
+    },
+    504: {
+        "raise_exception": LinkedInGatewayTimeoutError,
+        "message": "A gateway timeout occurred. There is a problem at LinkedIn's end."
+    }
+}
 
 def raise_for_error(response):
+    error_code = response.status_code
     try:
-        response.raise_for_status()
-    except (requests.HTTPError, requests.ConnectionError) as error:
-        try:
-            content_length = len(response.content)
-            if content_length == 0:
-                # There is nothing we can do here since LinkedIn has neither sent
-                # us a 2xx response nor a response content.
-                return
-            response = response.json()
-            if ('error' in response) or ('errorCode' in response):
-                message = '%s: %s' % (response.get('error', str(error)),
-                                      response.get('message', 'Unknown Error'))
-                error_code = response.get('status')
-                ex = get_exception_for_error_code(error_code)
-                if response.status_code == 401 and 'Expired access token' in message:
-                    LOGGER.error("Your access_token has expired as per LinkedIn’s security \
-                        policy. \n Please re-authenticate your connection to generate a new token \
-                        and resume extraction.")
-                raise ex(message)
-            else:
-                raise LinkedInError(error)
-        except (ValueError, TypeError):
-            raise LinkedInError(error)
-
+        response_json = response.json()
+    except Exception:
+        response_json = {}
+
+    if error_code == 404:
+        error_description = "The resource you have specified cannot be found."
```

Comment:
@harshpatel4crest, Since this, `if` the condition is generic among all streams, please keep the error description in the 'ERROR_CODE_EXCEPTION_MAPPING ' only